Create Date Formatter/Locator Reference

Examples of using the date formatters and locators is currently scattered haphazardly across the docs:
* [Dates documentation](https://matplotlib.org/api/dates_api.html?highlight=dates#module-matplotlib.dates)
* [Concise Date Formatter](https://matplotlib.org/gallery/ticks_and_spines/date_concise_formatter.html#sphx-glr-gallery-ticks-and-spines-date-concise-formatter-py)
* [Date tick labels](https://matplotlib.org/gallery/text_labels_and_annotations/date.html#sphx-glr-gallery-text-labels-and-annotations-date-py) 
* [Date Demo RRule](https://matplotlib.org/gallery/ticks_and_spines/date_demo_rrule.html#sphx-glr-gallery-ticks-and-spines-date-demo-rrule-py)
* [Date Index Formatter](https://matplotlib.org/gallery/ticks_and_spines/date_index_formatter2.html#sphx-glr-gallery-ticks-and-spines-date-index-formatter2-py)
* [Date Demo Convert](https://matplotlib.org/gallery/ticks_and_spines/date_demo_convert.html#sphx-glr-gallery-ticks-and-spines-date-demo-convert-py)

There currently isn't a good reference doc for the date formatters and locators.  It would be useful to have one example that's structurally/stylistically a combination of:
* [Locators Example](https://matplotlib.org/gallery/ticks_and_spines/tick-locators.html#sphx-glr-gallery-ticks-and-spines-tick-locators-py)
* [Formatters Example](https://matplotlib.org/gallery/ticks_and_spines/tick-formatters.html#sphx-glr-gallery-ticks-and-spines-tick-formatters-py)

@story645 Hi, I am running to errors trying to build the documentation in html format. I `cd`ed into `doc/` directory and ran `make html` but being running into the following errors:

`Makefile:30: recipe for target 'html' failed`
`make: *** [html] Error 2`
`

any guidance on how to properly build the docs would be really helpful. thanks :)

--- a/.flake8
+++ b/.flake8
@@ -120,6 +120,7 @@ per-file-ignores =
     examples/subplots_axes_and_figures/demo_constrained_layout.py: E402
     examples/text_labels_and_annotations/custom_legends.py: E402
     examples/ticks/date_concise_formatter.py: E402
+    examples/ticks/date_formatters_locators.py: F401
     examples/user_interfaces/embedding_in_gtk3_panzoom_sgskip.py: E402
     examples/user_interfaces/embedding_in_gtk3_sgskip.py: E402
     examples/user_interfaces/embedding_in_gtk4_panzoom_sgskip.py: E402

--- a/examples/ticks/date_formatters_locators.py
+++ b/examples/ticks/date_formatters_locators.py
@@ -1,0 +1,92 @@
+"""
+=================================
+Date tick locators and formatters
+=================================
+
+This example illustrates the usage and effect of the various date locators and
+formatters.
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.ticker as ticker
+from matplotlib.dates import (AutoDateLocator, YearLocator, MonthLocator,
+                              DayLocator, WeekdayLocator, HourLocator,
+                              MinuteLocator, SecondLocator, MicrosecondLocator,
+                              RRuleLocator, rrulewrapper, MONTHLY,
+                              MO, TU, WE, TH, FR, SA, SU, DateFormatter,
+                              AutoDateFormatter, ConciseDateFormatter)
+
+locators = [
+    ('AutoDateLocator(maxticks=8)', '2003-02-01', '%Y-%m'),
+    ('YearLocator(month=4)', '2003-02-01', '%Y-%m'),
+    ('MonthLocator(bymonth=[4,8,12])', '2003-02-01', '%Y-%m'),
+    ('DayLocator(interval=180)', '2003-02-01', '%Y-%m-%d'),
+    ('WeekdayLocator(byweekday=SU, interval=4)', '2000-07-01', '%a %Y-%m-%d'),
+    ('HourLocator(byhour=range(0,24,6))', '2000-02-04', '%H h'),
+    ('MinuteLocator(interval=15)', '2000-02-01 02:00', '%H:%M'),
+    ('SecondLocator(bysecond=(0,30))', '2000-02-01 00:02', '%H:%M:%S'),
+    ('MicrosecondLocator(interval=1000)', '2000-02-01 00:00:00.005', '%S.%f'),
+    ('RRuleLocator(rrulewrapper(freq=MONTHLY, \nbyweekday=(MO, TU, WE, TH,' +
+     ' FR), bysetpos=-1))', '2000-07-01', '%Y-%m-%d')
+]
+
+formatters = [
+    ('AutoDateFormatter(ax.xaxis.get_major_locator())'),
+    ('ConciseDateFormatter(ax.xaxis.get_major_locator())'),
+    ('DateFormatter("%b %Y")')
+]
+
+
+def plot_axis(ax, locator=None, xmax='2002-02-01', fmt=None, formatter=None):
+    """Set up common parameters for the Axes in the example."""
+    ax.spines.right.set_visible(False)
+    ax.spines.left.set_visible(False)
+    ax.spines.top.set_visible(False)
+    ax.yaxis.set_major_locator(ticker.NullLocator())
+    ax.tick_params(which='major', width=1.00, length=5)
+    ax.tick_params(which='minor', width=0.75, length=2.5)
+    ax.set_xlim(np.datetime64('2000-02-01'), np.datetime64(xmax))
+    if locator:
+        ax.xaxis.set_major_locator(eval(locator))
+        ax.xaxis.set_major_formatter(DateFormatter(fmt))
+    else:
+        ax.xaxis.set_major_formatter(eval(formatter))
+    ax.text(0.0, 0.2, locator or formatter, transform=ax.transAxes,
+            fontsize=14, fontname='Monospace', color='tab:blue')
+
+
+fig, ax = plt.subplots(len(locators), 1, figsize=(8, len(locators) * .8),
+                       layout='constrained')
+fig.suptitle('Date Locators')
+for i, loc in enumerate(locators):
+    plot_axis(ax[i], *loc)
+
+fig, ax = plt.subplots(len(formatters), 1, figsize=(8, len(formatters) * .8),
+                       layout='constrained')
+fig.suptitle('Date Formatters')
+for i, fmt in enumerate(formatters):
+    plot_axis(ax[i], formatter=fmt)
+
+
+#############################################################################
+#
+# .. admonition:: References
+#
+#    The use of the following functions, methods, classes and modules is shown
+#    in this example:
+#
+#    - `matplotlib.dates.AutoDateLocator`
+#    - `matplotlib.dates.YearLocator`
+#    - `matplotlib.dates.MonthLocator`
+#    - `matplotlib.dates.DayLocator`
+#    - `matplotlib.dates.WeekdayLocator`
+#    - `matplotlib.dates.HourLocator`
+#    - `matplotlib.dates.MinuteLocator`
+#    - `matplotlib.dates.SecondLocator`
+#    - `matplotlib.dates.MicrosecondLocator`
+#    - `matplotlib.dates.RRuleLocator`
+#    - `matplotlib.dates.rrulewrapper`
+#    - `matplotlib.dates.DateFormatter`
+#    - `matplotlib.dates.AutoDateFormatter`
+#    - `matplotlib.dates.ConciseDateFormatter`

